rust/datafusion/src/lib.rs

```diff
@@ -15,17 +15,46 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#![warn(missing_docs)]
+
 //! DataFusion is an extensible query execution framework that uses
 //! Apache Arrow as the memory model.
 //!
-//! DataFusion supports both SQL and a Table/DataFrame-style API for building logical query plans
+//! DataFusion supports both SQL and a DataFrame API for building logical query plans
 //! and also provides a query optimizer and execution engine capable of parallel execution
 //! against partitioned data sources (CSV and Parquet) using threads.
 //!
 //! DataFusion currently supports simple projection, selection, and aggregate queries.
-
-#![warn(missing_docs)]
-
+//!
+/// [ExecutionContext](../execution/context/struct.ExecutionContext.html) is the main interface
+/// for executing queries with DataFusion.
+///
+/// The following example demonstrates how to use the context to execute a query against a CSV
+/// data source using the DataFrame API:
+///
+/// ```
+/// # use datafusion::prelude::*;
```

Comment:
This example is 👍  -- I don't think I fully appreciated the `DataFrame` interface before.